tests/utils.go

```diff
@@ -423,6 +440,169 @@ func CreateSecret(name string, data map[string]string) {
 	}
 }
 
+func GetListOfManifests(pathToManifestsDir string) []string {
+	var manifests []string
+	files, err := ioutil.ReadDir(pathToManifestsDir)
+	if err != nil {
+		panic(err)
+	}
+	matchFileName := func(pattern, filename string) bool {
+		match, err := filepath.Match(pattern, filename)
+		if err != nil {
+			panic(err)
+		}
+		return match
+	}
+	isOpenshift := IsOpenShift()
+	for _, file := range files {
+		if matchFileName("*-for-ocp.yaml", file.Name()) {
+			if isOpenshift {
+				manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+			}
+		} else if matchFileName("*-for-k8s.yaml", file.Name()) {
+			if !isOpenshift {
+				manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+			}
+		} else if matchFileName("*.yaml", file.Name()) {
+			manifests = append(manifests, filepath.Join(pathToManifestsDir, file.Name()))
+		}
+	}
+	return manifests
+}
+
+func ReadManifestYamlFile(pathToManifest string) []unstructured.Unstructured {
+	// Read manifest
+	content, err := ioutil.ReadFile(pathToManifest)
+	PanicOnError(err)
+	// Split manifest by `---` to get single objects
+	fileAsString := string(content[:])
+	sepYamlfiles := strings.Split(fileAsString, "---")
```

Comment:
Maybe https://github.com/kubernetes/apimachinery/blob/master/pkg/util/yaml/decoder.go#L55 is also helpful. We have it in our vendor dependencies. It contains proper splitting logic.

Reply:
Changed to use k8s decoder